<!-- -->
<launch>
    <arg name="debug" default="false" doc="Debug mode" />

    <!-- <include file="$(find rplidar_ros)/launch/rplidar.launch" /> -->

    <node name="obstacle_detect" pkg="obstacle_detect" type="obstacle_detect_node" output="screen">
        <param name="debug" value="$(arg debug)"/>
        <param name="min_range" value="0.5"/>
        <param name="max_range" value="7"/>

        <param name="use_split_and_merge" value="true"/>
        <param name="detect_only_visible_circle" value="true"/>
        <param name="publish_seg_of_circle" value="true"/>

        <param name="min_group_points" value="5"/>

        <param name="max_group_distance" value="0.1"/>
        <param name="distance_proportion" value="0.00628"/>
        <param name="max_split_distance" value="0.2"/>
        <param name="max_merge_separation" value="0.2"/>
        <param name="max_merge_spread" value="0.2"/>
        <param name="max_circle_radius" value="0.6"/>
        <param name="radius_enlargement" value="0.3"/>

        <param name="frame_id" value="laser_frame"/>
        <param name="x_min" value="-0.2" />
        <param name="x_max" value="0.2" />
        <param name="y_max" value="0.5" />
    </node>
    <!-- <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" args="obstacle_detect" if="$(arg debug)"/> -->
    <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find obstacle_detect)/rviz/obstacle_detect.rviz" if="$(arg debug)"/>

</launch>
